scio-bigquery/src/main/scala/com/spotify/scio/bigquery/BigQueryITUtils.scala

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2016 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.bigquery
+
+import com.google.api.services.bigquery.model.{TableReference, TableSchema}
+import com.google.cloud.dataflow.sdk.io.BigQueryIO
+import com.google.cloud.dataflow.sdk.io.BigQueryIO.Write.WriteDisposition.WRITE_EMPTY
+import com.google.cloud.dataflow.sdk.io.BigQueryIO.Write.CreateDisposition.CREATE_IF_NEEDED
+import com.spotify.scio.bigquery.types.BigQueryType
+import com.spotify.scio.bigquery.types.BigQueryType.HasAnnotation
+
+import scala.collection.mutable.{Map => MMap}
+import scala.reflect.ClassTag
+import scala.reflect.runtime.universe._
+
+/**
+ * Utilities for BigQuery integration tests.
+ *
+ * Mix this in to test live queries against BigQuery service, using mock or sampled data.
+ */
+trait BigQueryITUtils {
+
+  val bq: BigQueryClient = BigQueryClient.defaultInstance()
+
+  private val mapping = MMap.empty[TableReference, TableReference]
+
+  /**
+   * Mock a BigQuery table. Each table can be mocked only once in a test class.
+   */
+  def mockTable(original: String): MockTable = mockTable(BigQueryIO.parseTableSpec(original))
+
+  /**
+   * Mock a BigQuery table. Each table can be mocked only once in a test class.
+   */
+  def mockTable(original: TableReference): MockTable = {
+    require(
+      !mapping.contains(original),
+      s"Table ${BigQueryIO.toTableSpec(original)} already registered for mocking")
+
+    val t = bq.getTable(original)
+    val temp = bq.temporaryTable(t.getLocation)
+    mapping += (original -> temp)
+    new MockTable(bq, t.getSchema, original, temp)
+  }
+
+  /**
+   * Perform a live query against BigQuery service, substituting mocked tables with test data.
+   */
+  def query(sqlQuery: String, flattenResults: Boolean = false): Seq[TableRow] = {
```

Comment:
Just a minor thing (and maybe it's just me) but would `runQuery()` or something be a better name? For me just `query()` makes it sound like a factory function to create some kind of query object instead of actually running one.